File: Java/Java10/addtime.java
import java.lang.Math;

class Time {
    public int hrs;
    public int mins;

    public Time(String strTime) {
        String[] parts = strTime.split(":");
        this.hrs = Integer.parseInt(parts[0]);
        this.mins = Integer.parseInt(parts[1]);
    }

    public Time(int hrs, int mins) {
        this.hrs = hrs;
        this.mins = mins;
    }

    public static Time elapse(Time s, int e) {
        int offset = (s.hrs * 60) + s.mins;
        int tot = offset + e;
        return new Time((int) Math.floor(tot / 60), tot % 60);
    }

    public String format() {
        return String.format("%02d:%02d", this.hrs, this.mins);
    }
}

class addtime {
    public static void usage() {
        System.out.println("Usage: <HH:MM> <mins>");
        System.exit(1);
    }

    public static void main(String[] argv) {
        if (argv.length < 2)
            usage();

        Time start = new Time(argv[0]);
        int elapse = Integer.parseInt(argv[1]);
        Time end = Time.elapse(start, elapse);

        String p1 = String.format("Start:\t%s\t%+d", start.format(), elapse);
        String p2 = String.format("End:\t%s", end.format());
        System.out.println(p1);
        System.out.println(p2);
    }
}

File: Nim/addtime.nim
import strformat
import strutils
import os
import parseopt
import options

type
  Minutes* = range[-59..59]
  Time* = object
    hrs*: int # top range, cannot have upper limit
    mins*: Minutes

proc abs*(t: Time): int =
  t.hrs * 60 + t.mins

proc `$`(t: Time): string =
  fmt"{t.hrs:02}:{t.mins:02}"

proc fromInt*(mins: int): Time =
  Time(hrs: mins /% 60, mins: mins mod 60)

proc fromString*(source: string): Time =
  if ':' in source:
    let lst = source.split(':', 1)
    return Time(hrs: lst[0].parseInt, mins: lst[1].parseInt)
  else:
    return source.parseInt.fromInt

proc `+`(t: Time, o: int): Time =
  let tot = t.abs + o
  fromInt(tot)

proc usage(status: int) =
  echo fmt"""Usage: {getAppFilename()} [-hql] <HH:MM> <mins | HH:MM>
  note: if mins_elapse is negative, precede it with '--'\n"
  options:
    -q quietly output end time
    -h print this message and exit
    -l print written language"""
  quit(status)

var
  quiet: bool = false
  start: Option[Time]
  elapse: Option[int]
  p = initOptParser()

while true:
  p.next()
  case p.kind
  of cmdEnd:
    if start.isNone or elapse.isNone:
      echo "Missing argument"
      usage(QuitFailure)
    break
  of cmdShortOption, cmdLongOption:
    case p.key[0]
    of 'h':
      usage(QuitSuccess)
    of 'l':
      echo fmt"Nim: {NimVersion}"
      quit(0)
    of 'q':
      quiet = true
    else:
      continue
  of cmdArgument:
    try:
      if start.isNone:
        start = some(fromString(p.key))
      elif elapse.isNone:
        elapse = some(fromString(p.key).abs)
    except ValueError:
      echo "Invalid argument"
      usage(QuitFailure)

if quiet:
  echo fmt"{start.get + elapse.get}"
else:
  echo fmt"""Start: {start.get} {elapse.get:+}
  End: {start.get + elapse.get}"""
